### ISSUE TITLE
fullscreen mode disables gamma correction

### ISSUE THREAD

**Issue by stephengold** [jMonkeyEngine/jmonkeyengine#1226]
I've noticed visually that applications with gamma correction enabled look very different in windowed display modes than they do in fullscreen. Visual comparisons suggest this is because gamma correction isn't actually enabled in fullscreen mode.

Seen in both v3.2.4 and v3.3.0-alpha5 of jMonkeyEngine with both v2 and v3 of LWJGL.

**Comment by riccardobl** [comment]
I can't reproduce this. Maybe there was something wrong in the code you were testing.

**Comment by stephengold** [comment]
I'll check again.

**Comment by stephengold** [comment]
I wasn't able to test on Linux due to issue #947. On Windows 7, a simple test with JME 3.3.2-stable and LWJGL 2.9.3 shows noticeable contrast between the viewport background and the stats background:

 1. in a 16bpp 640x480 window with `isGamma=true`
 2. in a 16bpp 640x480 fullscreen with `isGamma=true`
 3. in a 16bpp 640x480 fullscreen with `isGamma=false` 

but no contrast (black-and-black) in a 16bpp 640x480 window with `isGamma=false`.
 
Here is the test app:
```java
import com.jme3.app.SimpleApplication;
import com.jme3.font.BitmapText;
import com.jme3.math.ColorRGBA;

public class Main extends SimpleApplication {

    public static void main(String[] args) {
        Main app = new Main();
        app.start();
    }

    @Override
    public void simpleInitApp() {
        flyCam.setEnabled(false);
        
        ColorRGBA backgroundColor = new ColorRGBA(0.04f, 0.04f, 0.04f, 1f);
        viewPort.setBackgroundColor(backgroundColor);

        BitmapText statusText = new BitmapText(guiFont, false);
        statusText.setLocalTranslation(0f, cam.getHeight(), 0f);

        String message = 
                "   bpp=" + settings.getBitsPerPixel()
                + "   fullScreen=" + settings.isFullscreen() 
                + "   isGamma=" + settings.isGammaCorrection();
        statusText.setText(message);
        guiNode.attachChild(statusText);
    }
}
```

**Comment by riccardobl** [comment]
That's the normal behavior. Writing to an sRGB framebuffer causes the color to be gamma corrected, they will look different unless they are linearized beforehand 
```
        ColorRGBA backgroundColor = new ColorRGBA().setAsSrgb(0.04f, 0.04f, 0.04f, 1f);
```

**Comment by stephengold** [comment]
I agree that colors should look different between `isGamma=true` and `isGamma=false`, but that's not what this issue is about. The issue is that they *don't* look different in fullscreen, even between `isGamma=true` and `isGamma=false`.

**Comment by riccardobl** [comment]
I can't reproduce...could it be driver specific?
Do you get the same issue even with 24bpp  color depth?

**Comment by stephengold** [comment]
I suppose it could be a driver-specific issue.

I'm not finding any fullscreen modes with 24bpp color depth. However, I reproduced the issue with 32bpp 640x480 fullscreen.

**Comment by riccardobl** [comment]
Can you try to use native resolution fullscreen with lwjgl3?
Also, what drivers are you using?

**Comment by AdanosGotoman** [comment]
Hello everyone! May I ask you for help? In general, the situation is as follows: based on jme3 my friend from Germany created a program for editing world data for the game Gothic 3. As a separate module, he wrote a utility for viewing game levels and 3D models. Only there is a small problem - he didn't include the gizmo in it. I tried to get inspired by the original engine jme3, but I couldn't manage to add it for object manipulation. Looking at the original SceneEditTool class, I did not understand what was needed SceneComposerToolController and its fields and methods toolController. Can you to look into the class NodeViewerApp? I hope you can tell me how to implement a gizmo for this modification jme3.
https://github.com/georgeto/g3dit/tree/wip-jdk11

**Comment by stephengold** [comment]
```text
LWJGL 2.9.3 context running on thread jME3 Main
 * Graphics Adapter: nvd3dumx,nvwgf2umx,nvwgf2umx
 * Driver Version: 23.21.13.8873
 * Scaling Factor: 1
OpenGL Renderer Information
 * Vendor: NVIDIA Corporation
 * Renderer: GeForce GT 545/PCIe/SSE2
 * OpenGL Version: 4.6.0 NVIDIA 388.73
 * GLSL Version: 4.60 NVIDIA
 * Profile: Compatibility
```

I tested with LWJGL3 at the highest available resolutions and found good contrast:
+ in a 24 bpp 1914x1049 window with isGamma=true
+ in 32 bpp 1920x1080 fullscreen with isGamma=true

but poor contrast (black-and-black):
+ in a 24 bpp 1914x1049 window with isGamma=false
+ in 32 bpp 1920x1080 fullscreen with isGamma=false

In other words, the issue went away with high resolution and LWJGL3. Okay, that's interesting. Do you have a hypothesis that might explain all this?

**Comment by yaRnMcDonuts** [comment]
I can confirm I am also experiencing the same issue when trying to run full screen. 

Since @riccardobl was unable to reproduce the issue, I should note that my device's GPU is a NvidiaGeForce GTX 1080. I still need to test this on my older device with an AMD but will report back with the results when I do.

**Comment by stephengold** [comment]
@yaRnMcDonuts any news on this issue?

**Comment by Ali-RS** [comment]
I can not reproduce this issue. I am testing with Linux and JME 3.6.0-beta.

```
INFO: OpenGL Renderer Information
 * Vendor: Intel Open Source Technology Center
 * Renderer: Mesa DRI Intel(R) HD Graphics 3000 (SNB GT2)
 * OpenGL Version: 3.3 (Core Profile) Mesa 20.0.8
 * GLSL Version: 3.30
 * Profile: Core

``

**LWJGL 2**

fullscreen mode

![lwjgl2-fullscreen-mode](https://user-images.githubusercontent.com/5787647/218051225-fbd2e11b-d26b-4cef-b48a-6e62ea518d12.png)

window mode
![lwjgl2-window-mode](https://user-images.githubusercontent.com/5787647/218051245-9fe42fad-b380-49b1-9316-5ecb2f2ed9cf.png)

**LWJGL 3** 

fullscreen mode
![lwjgl3-fullscreen-mode](https://user-images.githubusercontent.com/5787647/218051568-64009518-f590-469a-b445-192389f7e099.png)

window mode
![lwjgl3-window-mode](https://user-images.githubusercontent.com/5787647/218051581-dd89dd67-280f-4f1a-98ce-0b2fb60f429a.png)

**Comment by Ali-RS** [comment]
And this is how it looks with gamma off:

![no-gamma](https://user-images.githubusercontent.com/5787647/218056913-89fd425c-dda7-4bb5-9394-fc697e52ddc2.png)

**Comment by riccardobl** [comment]
I've tested again on an RTX 2060s with 3.6.0-beta2 and i still can't reproduce :thinking:

**Comment by stephengold** [comment]
I reproduced the issue today, but it wasn't easy: I failed to repro it on Linux, nor could I repro it on my WIndows 11 laptop with an RTX 2070.

Here's what worked for me:

1. Boot the Windows 7 Professional SP 1 partition on my desktop (the machine with the NVIDA GeForce GT 545).
3. Check "Fullscreen?" in the display settings.
4. Visually compare the display, both with "Gamma correction" checked and unchecked.

After reproducing the issue (no visible difference between the displays) on 3.2.4-stable with LWJGL v2, I rebuilt the test app with 3.6.0-stable and jme3-awt-dialogs and ran it in the same environment. I was still able to repro the issue.

I was also able to reproduce the issue with LWJGL v3 instead of v2.

However, switching from fullscreen to windowed mode with LWJGL v2 made the issue go away: gamma correction had a visible effect.

My best guess would be it's a driver issue, but I can't prove it. The NVIDIA control panel displays the driver version as "388.73"
 
I'm leaving this issue open for now.
